File: JS-MongoDB/selectGenerator.js
var __dataframe = require("dataframe-js").DataFrame;
var __nosql = require("mongodb");
var __fs = require("fs")
var __parse = require("csv-parse");

const func = async () => {

	const __nosqlClient = new __nosql.MongoClient(
		"mongodb://127.0.0.1:27017/",
		{ useUnifiedTopology: true }
	);
	
	await __nosqlClient.connect();
	
	const nosql = __nosqlClient.db("mydb").collection("weather");

    var statement = "{ }"
	const select = async () => {
	
		let features = [];
		let objects = [];
		
		await nosql.find(JSON.parse(statement)).forEach((object) => {
			
			const keys = Object.keys(object);
			const n = features.length;
			let i;
			
			for(i = 0; i < n; ++i)
				if(!(JSON.stringify(features[i]) !== JSON.stringify(keys)))
					break;
			
			if(i === n) {
				features.push(keys);
				const __array = [];
				__array.push(object);
				objects.push(__array);
			} else
				objects[i].push(object);
			
		});
		
		let tables = [];
		
		for(i = 0; i < features.length; ++i)
			tables.push(new __dataframe(
				objects[i],
				features[i]
			));
			
		return tables;
	}
	
	var result = (await select()); // => TEST PASSED! -> Retrive list of dataframe from select operation.

	for(let table of result) {
		table = table.toArray();
		for(var __column in table) {
			var column = table[__column]
			console.log(column[2]) 
		}
	} // => TEST PASSED! -> Foreach on list of table, with set scope on table "Table".

    await __nosqlClient.close(); 
    
}

func(); 

File: PY-MongoDB/selectGenerator.py
#import os
#import random
#import time
#import math 
import pandas as pd
#import numpy as np
import json
#import urllib.request
#import pymysql
from pymongo import MongoClient

#import azure.functions as func
#from azure.storage.queue import QueueServiceClient

def main():
	#__environment = 'azure'
	#__queue_service = QueueServiceClient(account_url='https://"${storageName}".queue.core.windows.net', credential='"${storageKey}"')
	#__event = req.get_json()
	#data = __event['data']
	nosqlClient = MongoClient('mongodb://127.0.0.1:27017/') 
	nosqlDatabase = nosqlClient['mydb']
	nosql = nosqlDatabase['weather']
	statement =  '{ }' 
	def __selectQuery__():
	    result = nosql.find(json.loads(statement))
	    features = []
	    objects = []
	    for value in result:
	        feature = value.keys()
	        i = 0
	        for f in features:
	            if f == feature:
	                break
	            else:
	                i = i + 1 
	        if i + 1 > len(features):
	            features.append(feature)
	            objects.append([])
	            objects[len(objects) - 1].append(value)
	        else:
	            objects[i].append(value)
	
	    df = []
	    for i in range(len(features)):
	        df.append(pd.DataFrame(objects[i]))
	
	    return df
	
	result = __selectQuery__()
	
	for table in result:
		print(table)
	#__queue_service_client = __queue_service.get_queue_client('termination-"${function}"-"${id}"')
	#__queue_service_client.send_message('terminate')

main()